juju/bundle.py

```diff
@@ -681,9 +681,12 @@ def method():
             await context.model.create_offer(ep, offer_name=self.offer_name, application_name=application)
 
     def __str__(self):
+        endpoints = ""
+        if self.endpoints is not None:
+            endpoints = self.endpoints.join(",")
         return "create offer {offer_name} using {application}:{endpoints}".format(offer_name=self.offer_name,
                                                                                   application=self.application,
-                                                                                  endpoints=self.endpoints.join(","))
+                                                                                  endpoints=endpoints)
```

Comment:
Small nitpick here. If endpoints is empty the returned string will end with a colon. How about if you changed L686 to read `":" + self.endpoints.join(",")` and remove the colon from L687?

Reply:
Updated and fixed, good spot.